### PR TITLE
Adds some improvements to knowing if you've accidentally broken the d eclarations file generation

### PR DESCRIPTION
**Pull request by orta**
Figured changing the public API wasn't well documented enough, so I gave that a run-through and fixed up the DTS for TS users.

/cc @NMinhNguyen

### CODE REVIEW

File: scripts/create-danger-dts.ts [hunk 1]
```diff
@@ -13,4 +13,26 @@ fs.writeFileSync("source/danger.d.ts", output)
...
+      let { line, character } = diagnostic.file.getLineAndCharacterOfPosition(diagnostic.start!)
+      let message = ts.flattenDiagnosticMessageText(diagnostic.messageText, "\n")
+      console.error(`${diagnostic.file.fileName}:${line + 1}:${character + 1} - ${message}`)
+    } else {
+      console.error(`${ts.flattenDiagnosticMessageText(diagnostic.messageText, "\n")}`)
+    }
+  })
+} else {
+  console.log("Awesome - shipped to source/danger.d.ts, distribution/danger.d.ts and types/index.d.ts")
+}
```
Comment (orta) [inline scripts/create-danger-dts.ts]: Verifies the DTS compiles using the TS api

File: scripts/danger-dts.ts [hunk 2]
```diff
@@ -83,6 +83,9 @@ import * as GitHub from "@octokit/rest"
     .filter(line => {
       return !line.startsWith("import") && !line.includes("* @type ")
     })
+    .filter(line => {
+      return !line.includes("Please don't have")
+    })
```
Comment (orta) [inline scripts/danger-dts.ts]: This sentence used to be in a few DSL files, now it's in all of them

File: source/danger.d.ts [hunk 3]
```diff
@@ -200,10 +198,32 @@ interface BitBucketServerJSONDSL {
...
+  /** Make a post call against the bitbucket server API */
+  post(path: string, headers: any, body: any, suppressErrors?: boolean): Promise<any>
+
+  /** Make a put call against the bitbucket server API */
+  put(path: string, headers: any, body: any): Promise<any>
+
+  /** Make a delete call against the bitbucket server API */
+  delete(path: string, headers: any, body: any): Promise<any>
+}
+
```
Comment (orta) [inline source/danger.d.ts]: @NMinhNguyen - this is currently the only "exposed" API in the sense of auto-complete, the entire class instance is there with its real extra functions, you can expand this if there were specific functions you needed

**Comment by NMinhNguyen** [comment]
@orta Thanks so much for this. I tried to resolve conflicts in `CHANGELOG.md` directly on GitHub which created a merge commit which failed in CI. I then thought I would try `git rebase master` but that still failed. It's not obvious to me why Node 8 builds are timing out 😕

**Pull request merged**